api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirPatientDaoImpl.java

```diff
@@ -111,10 +123,55 @@ protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams
 				case FhirConstants.COMMON_SEARCH_HANDLER:
 					handleCommonSearchParameters(entry.getValue()).ifPresent(criteria::add);
 					break;
+				case FhirConstants.HAS_SEARCH_HANDLER:
+					entry.getValue().forEach(param -> handleHasAndListParam(criteria, (HasAndListParam) param.getParam()));
+					break;
 			}
 		});
 	}
 	
+	protected void handleHasAndListParam(Criteria criteria, HasAndListParam hasAndListParam) {
+		if (hasAndListParam != null) {
+			List<String> groupIds = new ArrayList<>();
+			hasAndListParam.getValuesAsQueryTokens().forEach(hasOrListParam -> {
+				hasOrListParam.getValuesAsQueryTokens().forEach(hasParam -> {
+					if (hasParam != null) {
+						String paramValue = hasParam.getParameterValue();
+						switch (hasParam.getTargetResourceType()) {
+							case FhirConstants.GROUP:
+								switch (hasParam.getReferenceFieldName()) {
+									case FhirConstants.INCLUDE_MEMBER_PARAM:
```

Comment:
Shouldn't we also have some sort of check for the `:id` part mentioned in the PR comments?